app/ante/eth.go

```diff
@@ -122,6 +123,13 @@ func (avd EthAccountVerificationDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx
 			return ctx, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "from address cannot be empty")
 		}
 
+		// check whether the sender address is EOA
+		fromAddr := common.BytesToAddress(from)
+		if codesize := avd.evmKeeper.GetCodeSize(fromAddr); codesize != 0 {
```

Comment:
actually, a better approach would be to query for the block hash and return an error if it's equal to the `EmptyCodeHash`

Reply:
thanks for your comment, i saw the comment from @albertchon . it will be more efficient to just check the code hash. 
i just think  using GetCodeSize would have better code encapsulation and readability.
I will change it to GetCodeHash.

Reply:
yeah, it's just that internally `GetCodeSize` queries for `GetCode` which itself queries `GetCodeHash`. So we can avoid a few db reads from just using the proposed method 👍